examples/fluids/qfunctions/densitycurrent.h

```diff
@@ -32,47 +32,6 @@
 #define M_PI    3.14159265358979323846
 #endif
 
-#ifndef setup_context_struct
```

Comment:
Any structs used in the QFunction must be defined in the QFunction source file or the Qfunctions cannot run on the GPU

Reply:
Yep! I just fixed it. Thanks but what is the reason?

Reply:
The entire QFunction source file is copied into the CUDA or HIP kernel that is generated at runtime, so those files need to contain every typedef the kernel will need to compile.